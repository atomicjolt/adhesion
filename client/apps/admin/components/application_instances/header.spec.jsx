import React from 'react';
import { shallow } from 'enzyme';
import Header from './header';

describe('application instances header', () => {
  let result;
  let props;
  let newInstance;

  beforeEach(() => {
    newInstance = false;
    props = {
      application: {
        name: 'test application',
      },
      newApplicationInstance: () => { newInstance = true; },
    };
    result = shallow(<Header {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the onClick function', () => {
    expect(newInstance).toBeFalsy();
    result.find('button').simulate('click');
    expect(newInstance).toBeTruthy();
  });
});
