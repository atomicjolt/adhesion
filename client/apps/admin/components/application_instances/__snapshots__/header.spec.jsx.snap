// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application instances header matches the snapshot 1`] = `
<div
  className="c-info"
>
  <div
    className="c-title"
  >
    <h1>
      test application
       Instances
    </h1>
  </div>
  <button
    className="c-btn c-btn--yellow"
    onClick={[Function]}
  >
    New Application Instance
  </button>
</div>
`;
